### PR TITLE
treewide: fetchgit -> fetchzip (git.kernel.org)

### PR DESCRIPTION
**Pull request by alyssais**
###### Description of changes

fetchzip is more efficient, because it doesn't do a full git clone, so it should be preferred where possible.

Where hashes have not been changed, I have verified that they don't need to be.  Where hashes have changed, in all cases this is because of .gitattributes files that exclude certain files from the tarball, and in these cases I have verified that the packages still build.

sbsigntool still uses fetchgit because it has a submodule, and ell and iwd still use fetchgit because git.kernel.org does not provide snapshot links for them.  I have contacted helpdesk@kernel.org to ask if this can be fixed.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alyssais** [comment]
> ell and iwd still use fetchgit because git.kernel.org does not provide snapshot links for them. I have contacted [helpdesk@kernel.org](mailto:helpdesk@kernel.org) to ask if this can be fixed.

Update: this is apparently on purpose, so we should continue to use fetchgit for these packages.

File: pkgs/development/python-modules/python-linux-procfs/default.nix [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 buildPythonPackage rec {
   pname = "python-linux-procfs";
   version = "0.6.3";
 
-  src = fetchgit {
-    url = "https://git.kernel.org/pub/scm/libs/python/${pname}/${pname}.git";
-    rev = "v${version}";
-    hash = "sha256-PPgMlL9oj4HYUsr444ZrGo1LSZBl9hL5SE98IASUpbc=";
+  src = fetchzip {
+    url = "https://git.kernel.org/pub/scm/libs/python/${pname}/${pname}.git/snapshot/${pname}-v${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-linux-procfs/default.nix:8]:
```suggestion
    url = "https://git.kernel.org/pub/scm/libs/python/python-linux-procfs/python-linux-procfs.git/snapshot/python-linux-procfs-v${version}.tar.gz";
```
nit

File: pkgs/os-specific/linux/tuna/default.nix [hunk 2]
```diff
@@ -16,10 +16,9 @@ buildPythonApplication rec {
   pname = "tuna";
   version = "0.15";
 
-  src = fetchgit {
-    url = "https://git.kernel.org/pub/scm/utils/${pname}/${pname}.git";
-    rev = "v${version}";
-    sha256 = "sha256-lRHlbdCQ0NcjcWgLvCze67kN8NsK0f5RmKfPbkHhk78=";
+  src = fetchzip {
+    url = "https://git.kernel.org/pub/scm/utils/${pname}/${pname}.git/snapshot/tuna-v0.15.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/tuna/default.nix:20]:
```suggestion
    url = "https://git.kernel.org/pub/scm/utils/tuna/tuna.git/snapshot/tuna-v${version}.tar.gz";
```

**Review by SuperSandro2000 — APPROVED** [review]
We do not have an updater for the snapshots, do we?

**Comment by SuperSandro2000** [comment]
> I assume it's the same situation as with static tarballs.

true, well that's definitely not blocking for this PR, if someone has motivation to fix it go ahead, otherwise feel free to merge when you think this is read

File: pkgs/os-specific/linux/tuna/default.nix [hunk 3]
```diff
@@ -16,13 +16,12 @@ buildPythonApplication rec {
   pname = "tuna";
   version = "0.15";
 
-  src = fetchgit {
-    url = "https://git.kernel.org/pub/scm/utils/${pname}/${pname}.git";
-    rev = "v${version}";
-    sha256 = "sha256-lRHlbdCQ0NcjcWgLvCze67kN8NsK0f5RmKfPbkHhk78=";
+  src = fetchzip {
+    url = "https://git.kernel.org/pub/scm/utils/tuna/tuna.git/snapshot/tuna-v0.15.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/tuna/default.nix:20]: Am I missing something or why are we not substituting the version number here?

Comment (alyssais) [reply pkgs/os-specific/linux/tuna/default.nix:20]: Ah, that's a mistake, thanks.

**Review by drupol — APPROVED** [review]
LGTM ! Well done :)

File: pkgs/os-specific/linux/tuna/default.nix (hunk 3 above)
Comment (alyssais) [reply pkgs/os-specific/linux/tuna/default.nix]: Fixed.

**Comment by SuperSandro2000** [comment]
and we waited to long and accumulated some merge conflicts. Also not sure if we should target staging here or not.

**Pull request merged**